core/src/main/java/greencity/service/impl/EcoNewsServiceImpl.java

```diff
@@ -60,8 +60,12 @@ public AddEcoNewsDtoResponse save(AddEcoNewsDtoRequest addEcoNewsDtoRequest,
                                       MultipartFile image, String email) {
         EcoNews toSave = modelMapper.map(addEcoNewsDtoRequest, EcoNews.class);
         toSave.setAuthor(userService.findByEmail(email));
-        if (image != null) {
-            toSave.setImagePath(fileService.upload(image).toString());
+        if (addEcoNewsDtoRequest.getImagePath() != null) {
+            toSave.setImagePath(addEcoNewsDtoRequest.getImagePath());
+        } else {
+            if (image != null) {
+                toSave.setImagePath(fileService.upload(image).toString());
+            }
         }
```

Comment:
Looks more readable as for me:
        String imagePath = addEcoNewsDtoRequest.getImagePath() == null && image != null
                ? fileService.upload(image).toString() : addEcoNewsDtoRequest.getImagePath();
        toSave.setImagePath(imagePath);